src/DbSelect.php

```diff
@@ -22,30 +24,10 @@ class DbSelect implements AdapterInterface
 
     /** @var Sql */
     protected $sql;
-
-    /**
-     * Database query
-     *
-     * @var Select
-     */
-    protected $select;
-
-    /**
-     * Database count query
-     *
-     * @var Select|null
-     */
-    protected $countSelect;
-
-    /** @var ResultSet */
-    protected $resultSetPrototype;
-
-    /**
-     * Total item count
-     *
-     * @var int
-     */
-    protected $rowCount;
+    protected Select $select;
+    protected ?Select $countSelect = null;
```

Comment:
```suggestion
    /** @var Select|null */
    protected $countSelect;
```
As @froschdesign already mentioned
> This is a BC break because the current class is not final and the properties are only protected.